src/shared_utilities/TimeVaryingInputs.jl

```diff
@@ -81,19 +81,54 @@ When passing single-site data
 When a `times` and `vals` are passed, `times` have to be sorted and the two arrays have to
 have the same length.
 
+=======
+When the input is a function, the signature of the function can be `func(time, args...;
+kwargs...)`. The function will be called with the additional arguments and keyword arguments
+passed to `evaluate!`. This can be used to access the state and the cache and use those to
+set the output field.
+
+For example:
+```julia
+CO2fromp(time, Y, p) = p.atmos.co2
```

Comment:
but maybe this isnt a good example? is this for the analytic time varying input case?